### ISSUE TITLE
Questions on Dropout Usage and Sigma Calibration in OOD Detection

### ISSUE THREAD

**Issue by jiusansan222** [mlaves/well-calibrated-regression-uncertainty#3]
Thank you for your work, it has been very helpful to me, but I have a few questions I need to ask you: 
1. Why is dropout set to true during the validation process in the train.py file? I think that dropout should only be enabled during the inference process for obtaining uncertainty with MC dropout. 
2. Does sigma calibration help with OOD (Out-Of-Distribution) detection? If uncertainties of all samples are multiplied by a fixed positive number, the relative size of the uncertainty values for OOD and ID (In-Distribution) samples will not change. Moreover, when I read the code, I found that the images drawn for OOD are also based on the results before sigma calibration, without involving sigma calibration
3. Recently, when conducting OOD detection on my dataset, I found that the overall uncertainty values for OOD samples are lower compared to in-domain samples, what do you think could be the reason? 
Thank you again for your answers

**Comment by mlaves** [comment]
Hi, thanks for your interest.
1. Ideally, you would keep dropout active all the time. At validation or test time, you would sample from the approximate posterior a couple of times in an MC integration fashion. In my training script, I skipped that at the validation step to speed-up the training loop, as I do not evaluate the uncertainty at the validation phase (only at test time).
2. Your assessment is correct. Sigma-scaling itself would not help with OOD detection, as it is a linear scale of the uncertainty.
3. I'm sorry, with the given information, I cannot guess the reason for that. In my experiments, uncertainty values for OOD samples were generally higher.
